### ISSUE TITLE
CA1801: When calling base constructor and expression body

### ISSUE THREAD

**Issue by stuartstein777** [dotnet/roslyn-analyzers#1776]
#### Analyzer package

Example: [Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)

assemblyref://Microsoft.CodeQuality.Analyzers

#### Package Version 

v2.6.1

#### Diagnostic ID

CA1801: Review unused parameters

#### Repro steps
Take the following class with a base class:

```
public sealed class Org : Entity<int>
    {
        private readonly string _name;
        public Org(int id, string name)
            : base(id) => _name = name;
        public string Name => _name;
    }

    public class Entity<T>
    {
        private readonly T _id;
        protected Entity(T id) => _id = id;
        public T Id => _id;
    }
```
Here `name` in the constructor of Org is being flagged as unused.

Changing it to not using expression body, the error goes away:

```
public sealed class Org : Entity<int>
    {
        private readonly string _name;
        public Org(int id, string name)
            : base(id)
        {
            _name = name;
        }
        public string Name => _name;
    }
```
^ This doesn't trigger error.
#### Expected behavior

Doesn't flag the parameter as unused

#### Actual behavior

Flags it as unused.

**Comment by mavasani** [comment]
Dupe of https://github.com/dotnet/roslyn-analyzers/issues/1606. Note that the issue was in the underlying compiler/analyzer API, which has been fixed in Visual Studio 15.8 release. Can you try the analyzer package against latest VS 15.8 Preview?
